dissect/target/filesystems/zip.py

```diff
@@ -0,0 +1,136 @@
+import io
+import logging
+import zipfile
+
+from dissect.util.stream import BufferedStream
+
+from dissect.target.exceptions import FileNotFoundError, FilesystemError
+from dissect.target.filesystem import Filesystem, VirtualFile, VirtualFilesystem
+from dissect.target.helpers import fsutil
+
+log = logging.getLogger(__name__)
+
+
+class ZipFilesystem(Filesystem):
+    """This Zip filesystem implementation is WIP and
+    has only been tested for opening java jar archives.
+    Its structure has been copied from the tar filesystem
+    implementation.
+    """
+
+    __fstype__ = "zip"
+
+    def __init__(self, fh, base=None, case_sensitive=True, zipinfo=None, *args, **kwargs):
+        fh.seek(0)
+
+        self.zip = zipfile.ZipFile(fh, mode="r")
+        self.base = base.strip("/") if base else ""
+
+        self._fs = VirtualFilesystem(case_sensitive=case_sensitive)
+
+        for member in self.zip.infolist():
+
+            if member.is_dir():
+                continue
+
+            mname = member.filename.strip("/")
+            if not mname.startswith(self.base):
+                continue
+
+            rel_name = mname[len(self.base) :]
+
+            # NOTE: Normally we would check here if the member is a symlink or not,
+            # however Python does not have symlink support in the zipfile module.
+            # see https://github.com/python/cpython/issues/82102
```

Comment:
```suggestion
            # NOTE: Normally we would check here if the member is a symlink or not
```
The explanation is now in the docstring, so only leave this note here.